### PR TITLE
ddl: allocate column ID and index ID in TableInfo.

### PR DESCRIPTION
**Pull request by coocood**
A row stores column IDs in varint encoding, smaller integer encodes to less bytes.
Use table info to allocate column ID to reduces disk usage.

### CODE REVIEW

File: ddl/foreign_key.go [hunk 1]
```diff
@@ -33,8 +33,9 @@ func (d *ddl) onCreateForeignKey(t *meta.Meta, job *model.Job) error {
 		job.State = model.JobCancelled
 		return errors.Trace(err)
 	}
+	tblInfo.MaxIndexID++
```
Comment (hanfei1991) [inline ddl/foreign_key.go]: Can we abstract these two lines as a function?

File: ddl/index.go [hunk 2]
```diff
@@ -153,11 +151,13 @@ func (d *ddl) onCreateIndex(t *meta.Meta, job *model.Job) error {
 	}
 
 	if indexInfo == nil {
-		indexInfo, err = buildIndexInfo(tblInfo, unique, indexName, indexID, idxColNames)
+		indexInfo, err = buildIndexInfo(tblInfo, unique, indexName, idxColNames)
 		if err != nil {
 			job.State = model.JobCancelled
 			return errors.Trace(err)
 		}
+		tblInfo.MaxIndexID++
```
Comment (hanfei1991) [inline ddl/index.go]: ditto.

**Comment by zimulala** [comment]
LGTM

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**